lib/hal/led.c

```diff
@@ -0,0 +1,15 @@
+#include <hal/led.h>
+
+void XSetAutoLED(void) {
+  HalWriteSMBusValue(0x20, SMC_REG_LEDMODE, FALSE, 0x00);
+}
+
+void XSetManualLED(int t1, int t2, int t3, int t4) {
+  HalWriteSMBusValue(0x20, SMC_REG_LEDMODE, FALSE, 0x01);
+
+  int colorScheme = t1<<1;
+  colorScheme = (colorScheme|t2)<<1;
+  colorScheme = (colorScheme|t3)<<1;
+  colorScheme = (colorScheme|t4) & 0xFF;
```

Comment:
This algorithm looks very broken.

Each LED color is 2 bits, not 1 bit. The & 0xFF is also strange.

It should probably be something like:

```c
//FIXME: make sure t1 - t4 are unsigned
assert(t1 <= 0x3);
assert(t2 <= 0x3);
assert(t3 <= 0x3);
assert(t4 <= 0x3);
uint8_t colorScheme = (t1 << 6) | (t2 << 4) | (t3 << 2) | (t4 << 0);
```

Reply:
https://github.com/XboxDev/nxdk/blob/a647c6d072beba61ac41fcb25a4a26de0dbd74c9/samples/led/main.c#L12 would suggest that each color is 1 bit per time value but constantly repeating?

Reply:
Your old code is right, it's lower and upper nibble. The code you originally submitted should work.
It should probably `assert((t1 & ~0x11) == 0)` to make sure that only the useful bits are being used.